src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -454,6 +467,53 @@ public bool TryCreateType(ITypeSymbol symbol, Func<System.Type, bool> validator,
             return true;
         }
 
+        private bool TryCreateTypeForIArrayTypeSymbol(ITypeSymbol symbol, [NotNullWhen(true)] out CSharpType? type)
+        {
+            type = null;
+            if (symbol is not IArrayTypeSymbol arrayTypeSymbol)
+            {
+                return false;
+            }
+
+            // For IArrayTypeSymbol, we can only handle it when the element type is a framework type.
+            var arrayType = TryGetFrameworkType(arrayTypeSymbol);
+            if (arrayType is not null)
+            {
+                type = new CSharpType(arrayType, false);
+                if (!type.IsValueType &&
+                    symbol.NullableAnnotation != NullableAnnotation.NotAnnotated)
+                {
+                    type = type.WithNullable(true);
+                }
+                return true;
+            }
+            return false;
+        }
+
+        private Type? TryGetFrameworkType(ISymbol namedTypeSymbol)
+        {
+            var fullMetadataName = GetFullMetadataName(namedTypeSymbol);
+            var fullyQualifiedName = $"{fullMetadataName}, {namedTypeSymbol.ContainingAssembly?.Name}";
+            return Type.GetType(fullMetadataName) ?? Type.GetType(fullyQualifiedName);
+        }
+
+        // There can be argument type missing
+        private bool TryPopulateArguments(ImmutableArray<ITypeSymbol> typeArguments, Func<Type, bool> validator, [NotNullWhen(true)] out CSharpType[]? arguments)
+        {
+            arguments = null;
+            var result = new List<CSharpType>();
+            foreach (var typeArgtment in typeArguments)
+            {
+                if (!TryCreateType(typeArgtment, validator, out CSharpType? type))
+                {
+                    return false;
+                }
+                result.Add(type);
+            }
+            arguments = result.ToArray();
+            return true;
+        }
```

Comment:
We could save a `ToArray` here if we have this PR merged in first: https://github.com/Azure/autorest.csharp/pull/4004
and we could do this:
```suggestion
        private bool TryPopulateArguments(ImmutableArray<ITypeSymbol> typeArguments, Func<Type, bool> validator, [NotNullWhen(true)] out IReadOnlyList<CSharpType> arguments)
        {
            argument = null;
            var result = new List<CSharpType>();
            foreach (var typeArgtment in typeArguments)
            {
                if (!TryCreateType(typeArgtment, validator, out CSharpType? type))
                {
                    return false;
                }
                result.Add(type);
            }
            arguments = result;
            return true;
        }
```

Reply:
and when we have the attribute `NotNullWhen`, we do not really need to add the question symbol on the type.

Reply:
If `arguments` is not nullable, how could you set it to null?

Reply:
> If `arguments` is not nullable, how could you set it to null?

The attribute does the magic.
For instance: https://github.com/Azure/autorest.csharp/blob/07e8ae69bdcd6d0b58d45d40f7ba38d8a6c37256/src/AutoRest.CSharp/Mgmt/Decorator/TypeExtensions.cs#L63
This is using a different attribute, but I believe it should follow the same rule